dkron/scheduler.go

```diff
@@ -72,39 +69,37 @@ func (s *Scheduler) Start(jobs []*Job, agent *Agent) error {
 }
 
 // Stop stops the scheduler effectively not running any job.
-func (s *Scheduler) Stop() {
+func (s *Scheduler) Stop() context.Context {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	ctx := s.Cron.Stop()
 	if s.started {
 		s.logger.Debug("scheduler: Stopping scheduler")
-		<-s.Cron.Stop().Done()
 		s.started = false
-		// Keep Cron exists and let the jobs which have been scheduled can continue to finish,
-		// even the node's leadership will be revoked.
-		// Ignore the running jobs and make s.Cron to nil may cause whole process crashed.
-		//s.Cron = nil
 
 		// expvars
 		cronInspect.Do(func(kv expvar.KeyValue) {
 			kv.Value = nil
 		})
 	}
 	schedulerStarted.Set(0)
+	return ctx
```

Comment:
Notice the difference between actively waiting for the jobs to end or not. Currently running jobs are not stopped on calling Stop(). They will continue execution until finishing.

We can not block execution on those callers, for example on leadership lost we can not wait an unknown amount of time that can be long.

We will only actively wait on graceful shutdown.

Reply:
Sure, I was just suggesting it might be helpful to write a comment in the places where jobs aren't awaited, so the design decision is clear in those places.